### ISSUE TITLE
comparisons with similar tools

### ISSUE THREAD

**Issue by pauliesnug** [oxalica/rust-overlay#200]
It would be helpful to see some comparisons with similar Nix tools, like Crane, nocargo, naersk, etc., with explainations on what projects would be benefited from the different solutions. Something like [this blogpost](https://www.tweag.io/blog/2022-09-22-rust-nix/), but more concise.

**Comment by oxalica** [comment]
I think I introduced what this project does in the first line of README.

> Pure and reproducible packaging of binary distributed rust toolchains. A compatible but better replacement for rust overlay of [nixpkgs-mozilla](https://github.com/mozilla/nixpkgs-mozilla), with also non-overlay and [Nix Flake](https://wiki.nixos.org/wiki/Flakes) interfaces (despite the project name).

Sorry if it's not that clear to your. Any improvements on the description are welcome.

> It would be helpful to see some comparisons with similar Nix tools, like Crane, nocargo, naersk, etc., with explainations on what projects would be benefited from the different solutions.

The projects you listed are not in the same scope of rust-overlay. They are the build-system to build rust projects in a Nix way. But rust-overlay are packageing Rust toolchains, that is, rustc, cargo and etc from stable, beta and nightly channels, for the build-system to perform the build. You may, for example, use rust-overlay's compiler and naersk's build system, which is mentioned in [your blog link](https://www.tweag.io/blog/2022-09-22-rust-nix/#naersk).

Our competing projects are:
- rustup: Official toolchain manager, no Nix/NixOS integration, easily break on NixOS update.
- nixpkgs's default rustc/cargo: Only limited to latest stable versions.
- [nixpkgs-mozilla](https://github.com/mozilla/nixpkgs-mozilla): At the last time I checked: requires IFD, non-deterministic, almost no flake support.
- [fenix](https://github.com/nix-community/fenix): Being developed independently around the same time, different interface, more lightweight but without historical nightly versions support (need manually providing hash when using `rust-toolchain.toml`). rust-overlay supports specifying any nightly version in recent 1-2years and easier `rust-toolchain.toml` integration, but comes with a tarball size disadvantage.

Hope this answers your question.

**Comment by pauliesnug** [comment]
Right, that makes sense. I'd also assume that a lot of these more large-scope buildsystem projects end up using `rust-overlay` under the hood. Thanks !! <3
